src/main/java/org/opensearch/security/filter/SecurityRestUtils.java

```diff
@@ -9,4 +32,28 @@ public static String path(final String uri) {
             return uri;
         }
     }
+
+    public static boolean shouldSkipAuthentication(HttpRequest request) {
+        String rawPath = path(request.uri());
+        String path = RestUtils.decodeComponent(rawPath);
+        Matcher matcher = PATTERN_PATH_PREFIX.matcher(path);
+        final String suffix = matcher.matches() ? matcher.group(2) : null;
+
+        boolean shouldSkipAuthentication = HttpMethod.OPTIONS.equals(request.method())
+            || HEALTH_SUFFIX.equals(suffix)
+            || WHO_AM_I_SUFFIX.equals(suffix);
+
+        return shouldSkipAuthentication;
+    }
+
+    public static boolean shouldSkipAuthentication(SecurityRequestChannel request) {
```

Comment:
This seems like it introduces duplicate code between these requests, can this duplication be removed?

Reply:
Removed the duplication and added tests